lib/services/plugins-service.ts

```diff
@@ -169,24 +172,69 @@ export class PluginsService implements IPluginsService {
 		return _.filter(nodeModules, nodeModuleData => nodeModuleData && nodeModuleData.isPlugin);
 	}
 
-	//This method will traverse all non dev dependencies (not only the root/installed ones) and filter the plugins.
 	public getAllProductionPlugins(projectData: IProjectData, dependencies?: IDependencyData[]): IPluginData[] {
-		const allProductionPlugins: IPluginData[] = [];
 		dependencies = dependencies || this.$nodeModulesDependenciesBuilder.getProductionDependencies(projectData.projectDir);
 
 		if (_.isEmpty(dependencies)) {
-			return allProductionPlugins;
+			return [];
 		}
 
-		_.forEach(dependencies, dependency => {
-			const isPlugin = !!dependency.nativescript;
-			if (isPlugin) {
-				const pluginData = this.convertToPluginData(dependency, projectData.projectDir);
-				allProductionPlugins.push(pluginData);
+		let productionPlugins: IDependencyData[] = dependencies.filter(d => !!d.nativescript);
+		productionPlugins = this.ensureValidProductionPlugins(productionPlugins, projectData.projectDir);
+		const pluginData = productionPlugins.map(plugin => this.convertToPluginData(plugin, projectData.projectDir));
+		return pluginData;
+	}
+
+	private ensureValidProductionPlugins = _.memoize<(productionDependencies: IDependencyData[], projectDir: string) => IDependencyData[]>(this._ensureValidProductionPlugins, (productionDependencies: IDependencyData[], projectDir: string) => {
+		let key = productionDependencies.map(p => p.directory).sort().join("\n");
```

Comment:
+version

Reply:
I've decided to stringify the whole array.